src/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -71,26 +71,32 @@ end
 
 function rrule(::typeof(cholesky), X::AbstractMatrix{<:Real})
     F = cholesky(X)
-    function cholesky_pullback(Ȳ)
-        ∂X = @thunk(chol_blocked_rev(Matrix(Ȳ), Matrix(F.U), 25, true))
+    function cholesky_pullback(Ȳ::Composite{<:Cholesky})
+        # TODO: no `unthunk`ing - https://github.com/JuliaDiff/ChainRulesCore.jl/issues/100
+        ∂X = if F.uplo === 'U'
+            @thunk(chol_blocked_rev(unthunk(Ȳ.U), F.U, 25, true))
+        else
+            @thunk(chol_blocked_rev(unthunk(Ȳ.L), F.L, 25, false))
+        end
         return (NO_FIELDS, ∂X)
     end
     return F, cholesky_pullback
 end
 
-function rrule(::typeof(getproperty), F::Cholesky, x::Symbol)
+function rrule(::typeof(getproperty), F::T, x::Symbol) where T <: Cholesky
     function getproperty_cholesky_pullback(Ȳ)
-        if x === :U
+        C = Composite{T}
+        ∂F = if x === :U
             if F.uplo === 'U'
-                ∂F = @thunk UpperTriangular(Ȳ)
+                C(U=(@thunk UpperTriangular(Ȳ)),)
             else
-                ∂F = @thunk LowerTriangular(Ȳ')
+                C(L=(@thunk LowerTriangular(Ȳ')),)
             end
         elseif x === :L
             if F.uplo === 'L'
-                ∂F = @thunk LowerTriangular(Ȳ)
+                C(L=(@thunk LowerTriangular(Ȳ)),)
             else
-                ∂F = @thunk UpperTriangular(Ȳ')
+                C(U=(@thunk UpperTriangular(Ȳ')),)
             end
```

Comment:
Because these only have a single field, there is no downside to swapping the thunk to the outside.

Reply:
done :)